### PR TITLE
ENH: Integrate with dust-extinction API

### LINKED ISSUE #132: Use separate package for extinction curve
As brought up by @karllark over at spacetelescope/pysynphot#54
<hr/>
There are external packages that provide extinction curves for python. One of these packages could be used instead of the numerical tables currently used.
Pros: offload the updating of the extinction curves to the "community"
Cons: Less control of the provided curves (but could be mitigated by automated testing against existing numerical results?)

Possible packages I know of:
https://github.com/kbarbary/extinction
https://github.com/karllark/dust_extinction
https://github.com/karllark/dust_attenuation
https://github.com/eblur/dust

In the interest of full disclose: I'm responsible for the 2nd package that has the aspiration of being an astropy affiliated package. The goals of my package are to provide dust extinction curves to the community and for use in dust extinction research (mine and others). Feedback is always welcomed.

...

I agree that only the dust_extinction package uses the astropy.modeling framework. I'm finding using this framework very powerful and useful. I've also been using the astropy.units framework. Very nice as well.

### PR DESCRIPTION
**Pull request by pllim**
Fix #132 

p.s. Sorry, Karl, no `__mul__` magic needed after all. 😉 

**TODO**

- [x] Address review comments.
- [x] Use Intersphinx to link to `dust-extinction` doc once karllark/dust_extinction#144 is merged and deployed.
- [ ] Open issue over at `astropy-tutorials` as a reminder to update Color Excess tutorial with this API in the future.

### CODE REVIEW

File: synphot/reddening.py [hunk 1]
```diff
@@ -74,11 +75,16 @@ def extinction_curve(self, ebv, wavelengths=None):
...
-        header = {
-            'E(B-V)': ebv,
-            'ReddeningLaw': self.meta.get('expr', 'unknown')}
+        x = self._validate_wavelengths(wavelengths)
+        header = {'E(B-V)': ebv}
+
+        # Duck-typing dust-extinction package API.
+        if HAS_DUST_EXTINCTION and hasattr(self.model, 'extinguish'):
+            y = self.model.extinguish(x, self.model.Rv * ebv)
+            header['ReddeningLaw'] = '{}'.format(self.model)
```
Comment (pllim) [inline synphot/reddening.py]: Put the short version in header, not the multi-line one.
```suggestion
            header['ReddeningLaw'] = '{!r}'.format(self.model)
```

File: synphot/reddening.py [hunk 2]
```diff
@@ -74,11 +75,16 @@ def extinction_curve(self, ebv, wavelengths=None):
...
-        y = 10 ** (-0.4 * self(x).value * ebv)
-        header = {
-            'E(B-V)': ebv,
-            'ReddeningLaw': self.meta.get('expr', 'unknown')}
+        x = self._validate_wavelengths(wavelengths)
+        header = {'E(B-V)': ebv}
+
+        # Duck-typing dust-extinction package API.
+        if HAS_DUST_EXTINCTION and hasattr(self.model, 'extinguish'):
+            y = self.model.extinguish(x, self.model.Rv * ebv)
```
Comment (karllark) [inline synphot/reddening.py]: Note that the extinguish function can take ebv as input.  so this line could be:
`y = self.model.extinguish(x, Ebv=ebv)`

Comment (pllim) [reply synphot/reddening.py]: Ah, thanks! I was simply following the tutorial example and didn't notice this option.

**Comment by pllim** [comment]
@karllark , is this sufficient? It looks like `dust-attenuation` API is different and not applicable here?

**Comment by karllark** [comment]
Looks great.
Yes, the `dust_attenuation` has a different API and is definitely in a more prototype state.  Probably need to spend time and get it into better shape before others might use it.  Having a similar API as `dust_extinction` would be a good idea.  Lots of other work needed for `dust_attenuation` as well.  Not pip installable for one.

**Comment by pllim** [comment]
Okay, good to know. Can you please formally approve if you are okay with this being merged? If not, please let me know what else is needed. Thank you!

**Pull request merged**